erpnext/projects/doctype/project/project.py

```
@@ -316,6 +319,17 @@ def update_purchase_costing(self):
 		total_purchase_cost = calculate_total_purchase_cost(self.name)
 		self.total_purchase_cost = total_purchase_cost and total_purchase_cost[0][0] or 0
 
+	def update_asset_capitalization_costing(self):
+		total_asset_capitalization_amount = frappe.db.sql(
+			"""select sum(total_value)
+			from `tabAsset Capitalization` where project = %s and docstatus=1""",
+			self.name,
+		)
+
+		self.total_asset_capitalization_amount = (
+			total_asset_capitalization_amount and total_asset_capitalization_amount[0][0] or 0
```

Ruff (0.15.5)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)